rocketpy/mathutils/function.py

```diff
@@ -3073,114 +3030,138 @@ def _check_user_input(
         >>> extrapolation
         'zero'
         """
+        if isinstance(source, Function):
+            source = source.get_source()
+
+        elif isinstance(source, (str, Path)):
+            # Read csv or txt files and create a numpy array
+            try:
+                source = np.loadtxt(source, delimiter=",", dtype=np.float64)
+            except ValueError:
+                with open(source, "r") as file:
+                    header, *data = file.read().splitlines()
+
+                header = [label.strip("'").strip('"') for label in header.split(",")]
+                source = np.loadtxt(data, delimiter=",", dtype=np.float64)
+
+                if len(source[0]) == len(header):
+                    if inputs == ["Scalar"]:
+                        inputs = header[:-1]
+                    if outputs == ["Scalar"]:
+                        outputs = [header[-1]]
+            except Exception as e:
+                raise ValueError(
+                    "Could not read the csv or txt file to create Function source."
+                ) from e
+
+        if isinstance(source, list):
+            # Triggers an error if source is not a list of numbers
+            source = np.array(source, dtype=np.float64)
+
+        if isinstance(source, np.ndarray):
+            inputs, interpolation, extrapolation = (
+                Function._validate_interpolation_and_extrapolation(
+                    inputs, interpolation, extrapolation, source
+                )
+            )
+            Function._validate_source_dimensions(inputs, outputs, source)
+
+        if isinstance(source, (int, float)):
+            # Convert number source into vectorized lambda function
+            temp = 1 * source
+
+            def source_function(_):
+                return temp
+
+            source = source_function
+        return source, inputs, outputs, interpolation, extrapolation
```

Comment:
The logic here checks if the source is a `Function` and gets its source, then goes on to validate that. However, that source has already been validated since it came from a Function object. 

Some redundant checks are happening here